MovieApp/MovieApp/App/Features/Movies/UseCases/MoviesUseCaseProtocol.swift

```diff
@@ -1,5 +1,5 @@
 protocol MoviesUseCaseProtocol {
     
-    func fetchMovies(completion: @escaping (NetworkResult<MoviesNetworkModel, NetworkError>) -> Void)
+    func fetchMovies(completion: @escaping (NetworkResult<[MovieModel], NetworkError>) -> Void)
```

Comment:
please use Result, because this no longer is only for networking. UC layer should not know where it got his data.
And also this should not be NetworkError but only Error.

Reply:
go through each layer and fix things with result and error